src/plopp/backends/matplotlib/line.py

```diff
@@ -175,7 +175,16 @@ def _preprocess_hist(self, data: dict) -> dict:
                 data["mask"]["y"] = np.concatenate(
                     (data["mask"]["y"][0:1], data["mask"]["y"]))
         if self._data.variances is not None:
-            data["variances"]["x"] = 0.5 * (x[1:] + x[:-1]) if hist else x
+            if hist:
+                if x.dtype.kind == 'M':
+                    xint = x.astype(int)
+                    xmid = (0.5 * (xint[1:] + xint[:-1])).astype(int)
+                    xvars = np.array(xmid, dtype=x.dtype)
+                else:
+                    xvars = 0.5 * (xint[1:] + xint[:-1])
+            else:
+                xvars = x
+            data["variances"]["x"] = xvars
```

Comment:
```suggestion
            data["variances"]["x"] = sc.midpoints(x) if hist else x
```
?

Reply:
thanks, didn't remember our `midpoints` worked for datetime. Also, because these are numpy arrays, I got stuck in thinking only about numpy arrays, but I will convert to Variable and back :+1:

Reply:
I wasn't aware that this is dealing in numpy arrays. Does this code matter for performance? That is, how often does this run when users, e.g. zoom into a plot or move a slider?

Reply:
I'm going to try and change it to convert to numpy later, so I can use Scipp for longer. It's not called on zoom, but it is called on slider interactions.